sources/mysql/README.md

```diff
@@ -1,83 +1,88 @@
+during data migration.
+
+For example, run
 
 ```sh
-harbourbridge -driver=mysqldump < my_mysqldump_file
+harbourbridge data -session=mydb.session.json -source=mysql -target-profile="instance=my-spanner-instance" < my_mysqldump_file
 ```
 
-To specify a particular Spanner instance to use, run:
+By default, HarbourBridge will generate a new Spanner database name to populate.
+You can override this and specify the database name to use by:
 
 ```sh
-mysqldump mydb | harbourbridge -driver=mysqldump -instance my-spanner-instance
+harbourbridge data -session=mydb.session.json -source=mysql -target-profile="instance=my-spanner-instance,dbname=my-spanner-database-name" < my_mysqldump_file
 ```
 
-By default, HarbourBridge will generate a new Spanner database name to populate.
-You can override this and specify the database name to use by:
+You can also run HarbourBridge in a oneshot mode, where it will perform both
+schema and data migration. This is useful for quick evaluation when source
+database size is small.
 
 ```sh
-mysqldump mydb | harbourbridge -driver=mysqldump -dbname my-spanner-database-name
+harbourbridge oneshot -source=mysql -target-profile="instance=my-spanner-instance" < my_mysqldump_file
 ```
 
 HarbourBridge generates a report file, a schema file, and a bad-data file (if
 there are bad-data rows). You can control where these files are written by
 specifying a file prefix. For example,
 
 ```sh
-mysqldump mydb | harbourbridge -prefix mydb. -driver=mysqldump
+harbourbridge schema -prefix=mydb. -source=mysql < my_mysqldump_file
 ```
 
 will write files `mydb.report.txt`, `mydb.schema.txt`, and
 `mydb.dropped.txt`. The prefix can also be a directory. For example,
 
 ```sh
-mysqldump mydb | harbourbridge -prefix ~/spanner-eval-mydb/ -driver=mysqldump
+harbourbridge schema -prefix=~/spanner-eval-mydb/ -source=mysql < my_mysqldump_file
 ```
 
 would write the files into the directory `~/spanner-eval-mydb/`. Note
 that HarbourBridge will not create directories as it writes these files.
 
 ### Directly connecting to a MySQL database
 
-To use the tool directly on a MySQL database called mydb, run
+In this case, HarbourBridge connects directly to the MySQL database to retrieve
+table schema and data. Set the `-source=mysql` and corresponding source profile
+connection parameters `host`, `port`, `user`, `db_name` and `password`.
+
+For example to perform schema conversion, run
 
 ```sh
-harbourbridge -driver=mysql
+harbourbridge schema -source=mysql -source-profile="host=<>,port=<>,user=<>,db_name=<>"
 ```
 
-It is assumed that _MYSQLHOST_, _MYSQLPORT_, _MYSQLUSER_, _MYSQLDATABASE_ environment
-variables are set. Password can be specified either in the _MYSQLPWD_ environment
-variable or provided at the password prompt.
+Parameters `port` and `password` are optional. Port (`port`) defaults to `3306`
+for MySQL source. Password can be provided at the password prompt.
+
+(⚠ Deprecated ⚠) Set environment variables `MYSQLHOST`, `MYSQLPORT`,
```

Comment:
Not suggest a change here, but just noting our plan that we should deprecate the MySQL environment variables because they are non-standard.
But we should keep the Postgres ones, because they are standard (and commonly used).

Reply:
Acknowledged.